src/pvrclient-mediaportal.cpp

```diff
@@ -2481,3 +2481,70 @@ PVR_ERROR cPVRClientMediaPortal::GetStreamReadChunkSize(int& chunksize)
   chunksize = 32 * 1024;
   return PVR_ERROR_NO_ERROR;
 }
+
+namespace
+{
+    int GetFileContents(const std::string& url, std::string& content)
+    {
+      content.clear();
+      kodi::vfs::CFile file;
+      if (file.OpenFile(url))
```

Comment:
I don't know the Kodi vfs API, but usually you have to close the filter once you have finished reading, no?

Reply:
No, it auto closes once it's out of scope.

Reply:
https://github.com/xbmc/xbmc/blob/dbc793ec9106485c71c2c9d397c22bbd81c44c63/xbmc/addons/kodi-dev-kit/include/kodi/Filesystem.h#L1843